src/snowflake/snowpark/session.py

```diff
@@ -643,11 +639,6 @@ def auto_clean_up_temp_table_enabled(self) -> bool:
         When setting this parameter to ``True``, Snowpark will automatically clean up temporary tables created by
         :meth:`DataFrame.cache_result` in the current session when the DataFrame is no longer referenced (i.e., gets garbage collected).
         The default value is ``False``.
-
-        Note:
```

Comment:
We can still provide this guarantee, by checking all entries of the count map whether a count reaches 0. But given we're not using watch thread, I don't think this guarantee makes much sense. We only need to do our best effort to clean up temp tables when this parameter is enabled. Alternatively, we can always easily add this guarantee later when the customer requests it.

Reply:
This behavior is irrelevant with the threading work, it is a behavior of the cleaner. if we do not have this, what we are saying is we only clean up temp tables whose reference reach 0 after the cleaner is started. let's make sure this is documented clearly somewhere

Reply:
We can say the temporary tables will only be dropped when this parameter is turned on during garbage collection, whereas the garbage collection in Python is triggered opportunistically and the timing is not guaranteed. I think it's also clear. Let me know wdyt.